### DISCUSSION TITLE
Cannot change virtual display screen resolution using xvfb-metrics parameter

### DISCUSSION THREAD

**Discussion by jfhampton** [seleniumbase/SeleniumBase#3176] [General]
Hi,

I am using xvfb on linux ubuntu server 24.04 in UC mode with SeleniumBase 4.31.3.
I cannot get screen resolution to change when using parameter `--xvfb-metrics` to launch the test: screen resolution is always `1366x768`.

Test is launched with following pamateters: `--xvfb --xvfb-metrics="1920,1080"`
UC mode is activated using context manager: `with SB(uc=True, user_data_dir="my_user_dir_1") as sb:`

I have traced the execution of SeleniumBase code and I see that `setUp` function is called after `__activate_virtual_display_as_needed` (in `seleniumbase/fixtures/base_case.py`).

Reading `--xvfb-metrics` parameter occurs in `setUp` function: `__activate_virtual_display_as_needed` gets empty `self._xvfb_width` and `self._xvfb_height` attributes and in result sets the default resolution (`1366x768`).

Am I using the xvfb parameters wrong ?

**Comment by mdmintz** [comment]
Nice catch. Thanks for reporting this.
Fixed in `4.31.4` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.31.4

**Comment by jfhampton** [comment]
Thanks !
I have tested `4.31.4`: resolution provided with parameter `xvfb-metrics` is applied to virtual display.
